pkg/ipam/crd.go

```diff
@@ -128,12 +128,8 @@ func newNodeStore(nodeName string, conf Configuration, owner Owner, k8sEventReg
 				defer func() { k8sEventReg.K8sEventReceived("CiliumNode", "update", valid, equal) }()
 				if oldNode, ok := oldObj.(*ciliumv2.CiliumNode); ok {
 					if newNode, ok := newObj.(*ciliumv2.CiliumNode); ok {
-						if oldNode.DeepEqual(newNode) {
-							equal = true
-							return
-						}
 						valid = true
```

Comment:
It looks like there was another bug in the old code here: `valid` was only set to `true` if `oldNode` and `newNode` were not equal. This PR looks like it fixes that.